include/albatross/src/models/ransac.hpp

```diff
@@ -186,91 +195,66 @@ ransac(const RansacFunctions<FitType, GroupKey> &ransac_functions,
   std::size_t i = 0;
   std::size_t failed_candidates = 0;
   while (i < max_iterations) {
+
+    output.iterations.emplace_back(RansacIteration<GroupKey>());
+    RansacIteration<GroupKey> &iteration = output.iterations.back();
+
     // Sample a random subset of the data and fit a model.
-    auto candidate_groups =
+    iteration.candidates =
         random_without_replacement(groups, random_sample_size, gen);
 
-    if (DEBUG_RANSAC) {
-      std::cout << "=============================================== " << i
-                << std::endl;
-      std::cout << "Candidates:" << std::endl;
-      for (const auto &candidate : candidate_groups) {
-        std::cout << "  - ";
-        print_group_or_index(candidate, groups);
-        std::cout << std::endl;
-      }
-    }
-
     // Sometimes it's hard to design an inlier metric which is
     // reliable if the candidate groups are tainted with outliers.
     // Consider a situation where there are multiple correlated
     // outliers and one of those ends up in the candidate set, it's
     // possible the model will then reasonably predict inliers
     // AND outliers resulting in a large consensus set.  This
     // is_valid_candidate step allows you to filter those cases out.
-    if (!ransac_functions.is_valid_candidate(candidate_groups)) {
+    if (!ransac_functions.is_valid_candidate(iteration.candidates)) {
       ++failed_candidates;
-      if (DEBUG_RANSAC) {
-        std::cout << "Invalid Candidate" << std::endl;
-      }
       if (failed_candidates >= max_failed_candidates) {
         output.return_code = RANSAC_RETURN_CODE_EXCEEDED_MAX_FAILED_CANDIDATES;
         return output;
       }
       continue;
     }
 
-    const auto fit = ransac_functions.fitter(candidate_groups);
+    const auto fit = ransac_functions.fitter(iteration.candidates);
 
     // Any group that's part of the candidate set is automatically an inlier.
-    std::vector<GroupKey> candidate_consensus = candidate_groups;
-    std::vector<GroupKey> outliers;
-
-    if (DEBUG_RANSAC) {
-      std::cout << "Inliers:" << std::endl;
-    }
+    std::vector<GroupKey> candidate_consensus = iteration.candidates;
     // Find which of the other groups agree with the reference model
     // which gives us a consensus (set of inliers).
     for (const auto &possible_inlier : groups) {
-      if (!contains_group(candidate_groups, possible_inlier)) {
+      if (!contains_group(iteration.candidates, possible_inlier)) {
         double metric_value =
             ransac_functions.inlier_metric(possible_inlier, fit);
         const bool accepted = metric_value < inlier_threshold;
         if (accepted) {
-          candidate_consensus.emplace_back(possible_inlier);
+          iteration.inliers[possible_inlier] = metric_value;
```

Comment:
I think it's a bit better to use `iteration.inliers.insert({possible_inlier, metric_value});` here.
That skips searching for prior existence of the key in the map (at the cost of doing nothing if it exists)